### DISCUSSION TITLE
Availability of PrimeView web application

### DISCUSSION THREAD

**Discussion by rbergen** [PlummersSoftwareLLC/Primes#560] [Announcements]
Hi all,

I'm writing this to inform you that as of now, a web application is available with which benchmark results of the current set of solutions can be retrieved, viewed, sorted and filtered. It's available here: https://plummerssoftwarellc.github.io/PrimeView/.

Currently, the benchmark reports shown by the application are generated daily on the same system that Dave uses for the comparisons in his video series on YouTube. In the future, we want to allow others to upload their benchmark results as well, but this will require more work. We don't yet know if and when we'll be able to add this capability, but we will definitely announce it if and when it becomes available.

The application allows you to define your own filter presets and it will remember these (and other settings) across sessions within the same browser.

If you have any suggestions (PRs) for or issues with the application, then please **open them on the application's own repo**, that being https://github.com/PlummersSoftwareLLC/PrimeView. Do expect slower response times on PrimeView PRs and issues than here, because we prioritize the main project (captured in this repo) over the support project that is PrimeView.

We hope that the application makes a positive contribution to your participation in the project!

Best,

the Primes maintainers

**Comment by fvbakel** [comment]
Great work! Great to see the results on his machine. No wonder he keeps telling that C or assembly are not the fastest. I get a complete different ranking on my hardware.

**Reply by marghidanu** [reply]
We are looking to expand the data collection from several machines, right now we have this one. I’ll add Raspberry Pi in the following days.

**Reply by fvbakel** [reply]
@marghidanu can't wait to get the results of that.

**Reply by louie-github** [reply]
I was also quite surprised at the results. It's completely different from my hardware. Although, he is using an old 2013 Celeron chip while I have a Core i5 9th generation, so it's not exactly an apples-to-apples comparison :>

Also very much excited for other machines' results!

**Comment by louie-github** [comment]
Maybe you could consider adding a link to the PrimeView web app to the README in addition to the "website" part of this repository so it'd be more visible to everyone.

Thanks for your hard work!

**Reply by rbergen** [reply]
You're very welcome. Concerning PrimeView: I had a lot of fun writing that, and I'm thrilled that it's "finally" being used by others as well. :)

Adding a link to the repo README is actually not a bad idea. I might open a tiny PR for that, sometime soon.

**Comment by mayerrobert** [comment]
The PrimeView webapp is great, however for some reason the results of PrimeLisp/solution_2 don't show up?! I'm pretty sure that that's not an error within the webapp because PrimeLisp/solution_2 results are not in the generated JSON files on https://primes.fra1.digitaloceanspaces.com/.

Either there is an error on my side or something breaks during the benchmark run but I'm not able to figure out what went wrong.

What to do?

**Reply by rbergen** [reply]
I took a look on the benchmark system, and there is a small issue with your Dockerfile, that exhibits on that machine. Your Dockerfile currently looks as follows:

```dockerfile
FROM daewok/sbcl:2.1.6-alpine3.13

WORKDIR /opt/app

COPY *.lisp .
COPY run.sh .

RUN chmod +x run.sh

ENTRYPOINT [ "./run.sh" ]
```

The first `COPY` directive copies multiple files, which means the target needs to end with a slash (/). If you were to open a PR to change that line to 
```dockerfile
COPY *.lisp ./
```
then Lisp/solution_2 should show up in the first report generated after that PR is merged.

**Reply by mayerrobert** [reply]
Thanks for the fast response!

I tested the stuff with podman on Fedora 33 which *should* behave like docker but oh well.

Thanks again! Even tough it will be an easy fix I would never have figured that out myself.

**Reply by rbergen** [reply]
No worries. It's a really weird discrepancy between the standard behaviour of Docker Desktop (at least on Windows) and Docker Engine (at least on Linux) as well. Docker Desktop processes Dockerfiles with this problem without complaining, Docker Engine aborts with an error.

**Reply by rbergen** [reply]
@mayerrobert I just took a look at the report that was generated this morning, and "on my machine" Lisp/solution_2 does show up now.
